src/Services/SolrCoreService.php

```diff
@@ -311,17 +320,16 @@ public function getSolrVersion($handler = null): int
         $result = $client->get('solr/admin/info/system?wt=json', $clientOptions);
         $result = json_decode($result->getBody(), 1);
 
-        $version = 7;
-        // Newer than 4, older than 7, a few new features added
-        if (version_compare('6.9.9', $result['lucene']['solr-spec-version']) >= 0) {
-            $version = 5;
-        }
-        // Old version 4
-        if (version_compare('4.9.9', $result['lucene']['solr-spec-version']) >= 0) {
-            $version = 4;
+        foreach (static::$solr_versions as $version) {
+            $compare = version_compare($version, $result['lucene']['solr-spec-version']);
+            if ($compare >= 0) {
```

Comment:
:wrench: Nice one. Correct me here, I think we are looking for `$compare` having `-1|0`
```suggestion
            if ($compare < 1) {
```

Reply:
Yeah, `!== 1`,  `< 1` and `>= 0`, it's all doing the same indeed. I went with `>= 0` from the docs...

I think `!== 1` is the most logically readable, as it would explicitly say "The version on the left must not be higher than the version compared to on the right"

Reply:
Yup I agree with `$compare !== -1`

Reply:
Just let me test and figure out if it's `-1` or `1` for sure, and I'll update the PR :)

Reply:
No worries check `0` as well :)

Reply:
Updated, after writing out the docblock, I'm certain it should skip if `-1` :)